### ISSUE TITLE
noImplicitOverride reports in JS files with checkJs enabled, can only ts-ignore the line

### ISSUE THREAD

**Issue by jakebailey** [microsoft/TypeScript#45591]
# Bug Report

### 🔎 Search Terms

override checkjs jsdoc

### 🕗 Version & Regression Information

- This changed between versions 4.2 and 4.3

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?noImplicitOverride=true&filetype=js#code/MYGwhgzhAEBiD29oG8CwAoa0C2BTALgBbwAmAFAJQoZZYBOBArnQHbQCMATAMwAsN0AL4Zh6DKEgwAQmDrRcAD3y4WJGAiRpMOAsXJUttaA3zM2ERgAdcdAHR4ipStADU0XpwDcA0YKA)

### 💻 Code

```tsx
class Foo {
  method() {
    return 1234
  }
}

class Bar extends Foo {
  method() {
    return super.method() + 42;
  }
}
```

### 🙁 Actual behavior

In my projects, I set `checkJs` to enforce that our non-TS files also typecheck. We also use the `noImplicitOverride` rule.

But, if you try to create a class with method overrides in JS files, you get the override errors, even though there doesn't appear to be a way to silence them except via `// @ts-ignore`.

### 🙂 Expected behavior

`noImplicitOverride` is not reported in JS files (or, there's some JSDoc way to add it, like types can be).

**Comment by a-tarasyuk** [comment]
The issue is labeled "Domain: Error Messages". What should be the expected error message?

/cc @DanielRosenwasser

**Comment by DanielRosenwasser** [comment]
> What should be the expected error message?

@sandersn might have opinions but

```
This member must have a leading JSDoc comment with an `@override` tag because it overrides a member in the base class 'Foo'.
```

> TypeScript does not have good enough tools to handle jsdoc/comments yet. It's hard to add a jsdoc tag to something if there's already exists leading comments.

This is a good point. We would accept a PR that doesn't include a quick fix, but maybe Nathan also has some ideas there.

**Comment by Kingwl** [comment]
Here's a PR to try to add quickfix support in js file. https://github.com/Kingwl/TypeScript/pull/8/files.

**Comment by sandersn** [comment]
> It's hard to add a jsdoc tag to something if there's already exists leading comments.

I believe inferFromUsage has code for this.

**Comment by DanielRosenwasser** [comment]
@Kingwl see if there's anything there you might be able to leverage for your current work, and let us know if you run into any issues.

**Comment by Kingwl** [comment]
I guess it's might helpful to @stkevintan.
